### PR TITLE
[ new ] map-concat

### PR DESCRIPTION
**Pull request by gallais**
Preparing next year's coursework and I'll need this. :)

### CODE REVIEW

**Review by JacquesCarette — APPROVED** [review]
Looks good.

Makes me sad that `concat` isn't defined in terms of `fold` so that this could become a special case of `map-fold`. Ah well.

**Review by jamesmckinna — APPROVED** [review]

**Comment by jamesmckinna** [comment]
> Looks good.
> 
> Makes me sad that `concat` isn't defined in terms of `fold` so that this could become a special case of `map-fold`. Ah well.

Two things 
* the library seemed to prefer direct recursive definitions over higher-order combinators but this (more didactic?) style seems to have gradually been refactored away
* for `Vec` it's only relatively recently cf. #1668 / #942 that we've had systematic `foldr`/`foldl` fusions

So: possible issue (`v3.0`? `hypothetical-rewrite`?) to systematically refactor the whole lot?

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,3 +26,8 @@ New modules
 
 Additions to existing modules
 -----------------------------
+
+* New lemma in `Data.Vec.Properties`:
+  ```agda
+  map-concat : map f (concat xss) ≡ concat (map (map f) xss)
```
Comment (jamesmckinna) [inline CHANGELOG.md:32]: aaaargh! This type has the opposite orientation to
```agda
Data.List.Properties.concat-map : ∀ {f : A → B} → concat ∘ map (map f) ≗ map f ∘ concat
```

Suggest reversing the orientation for `Vec`, for consistency's sake...

**Review by jamesmckinna — CHANGES_REQUESTED** [review]
Aaargh... my previous `Approve` was too hasty...

File: CHANGELOG.md (hunk 1 above)
Comment (jamesmckinna) [reply CHANGELOG.md:32]: Or else: a `breaking` refactoring of the `List` definition... :-(

Comment (jamesmckinna) [reply CHANGELOG.md:32]: Aaargh... ignore! `map-concat` vs. `concat-map`... sigh :man_facepalming:

**Review by jamesmckinna — APPROVED** [review]
Third time's the charm...

**Pull request merged**